Context slides (#88)
component-driven/component-driven-development: slides/resilient-component-libraries.mdx

---

> > 20 minutes

Reviewer: I liked the hourglass emoji 😢

Author: Me too but it looks ugly in italics

---

```jsx
import React, { useCallback } from 'react';
import { useDropzone } from 'react-dropzone';

function MyDropzone() {

> 	const onDrop = useCallback(acceptedFiles => {

Reviewer: ```suggestion
	const onDrop = React.useCallback(acceptedFiles => {
```

For consistency with other slides?

		// Do something with the files
	}, []);
	const { getRootProps, getInputProps, isDragActive } = useDropzone({ onDrop });

	return (
		<div {...getRootProps()}>
			<input {...getInputProps()} />
			{isDragActive ? (
				<p>Drop the files here ...</p>
			) : (

> 				<p>Drag 'n' drop some files here, or click to select files</p>

Reviewer: ```suggestion
				<p>Drag'n'drop some files here, or click to select files</p>
```

---

> ## React.Context

Reviewer: ```suggestion
## React Context
```

> - global state store (you don't need Redux!)

Reviewer: ```suggestion
- global state store (you may not need Redux!)
```

---

<Primary>

> # React.Context

Reviewer: ```suggestion
# React Context
```

</Primary>

---

> Context is designed to share data that can be considered “global” for a tree of React components

---

<SplitTitle>

## Avoid "prop drilling"

```jsx
import { ThemeProvider } from 'styled-components';
import theme from '../theme';

<ThemeProvider theme={theme}>
	<App />
</ThemeProvider>;
```

</SplitTitle>

---

> ## Avoid re-renders by passing React reconciliation

Reviewer: Bypassing? I'm not sure what “passing React reconciliation” means.

Author: yes! thanks!

[React at 60fps](https://medium.com/hackernoon/react-at-60fps-4e36b8189a4c)

---

> <Demo src="http://react-sync-scroll.netlify.com" />

Reviewer: ```suggestion
<Demo src="https://react-sync-scroll.netlify.app/" />
```
